src/V3EmitCModel.cpp

```diff
@@ -422,7 +422,7 @@ class EmitCModel final : public EmitCFunc {
         }
         puts("Verilated::endOfEval(vlSymsp->__Vm_evalMsgQp);\n");
 
-        if (v3Global.opt.profExec()) puts("VL_EXEC_TRACE_ADD_RECORD(vlSymsp).evalEnd();\n");
+        if (v3Global.opt.profExec()) puts("VL_EXEC_TRACE_ADD_RECORD(vlSymsp).sectionPop();\n");
```

Comment:
Maybe have sectionPush be wrapped in a class which calls it, then when destroyed automatically calls sectionPop?  (Likewise cleans up code elsewhere).

Reply:
I agree that would be nice, but it does not work nicely for all use cases. Currently `eval` have multiple push/pop, with some pop not strictly on scope exit:

```
   void Vt_gantt___024root___eval(Vt_gantt___024root* vlSelf) {
       if (false && vlSelf) {}  // Prevent unused
       Vt_gantt__Syms* const __restrict vlSymsp VL_ATTR_UNUSED = vlSelf->vlSymsp;
       VL_DEBUG_IF(VL_DBG_MSGF("+    Vt_gantt___024root___eval\n"); );
       // Init
       VlTriggerVec<1> __VpreTriggered;
       IData/*31:0*/ __VnbaIterCount;
       CData/*0:0*/ __VnbaContinue;
       // Body
       VL_EXEC_TRACE_ADD_RECORD(vlSymsp).sectionPush("loop nba");
       __VnbaIterCount = 0U;
       __VnbaContinue = 1U;
       while (__VnbaContinue) {
           __VnbaContinue = 0U;
           vlSelf->__VnbaTriggered.clear();
           VL_EXEC_TRACE_ADD_RECORD(vlSymsp).sectionPush("loop act");
           vlSelf->__VactIterCount = 0U;
           vlSelf->__VactContinue = 1U;
           while (vlSelf->__VactContinue) {
               vlSelf->__VactContinue = 0U;
               Vt_gantt___024root___eval_triggers__act(vlSelf);
               if (vlSelf->__VactTriggered.any()) {
                   vlSelf->__VactContinue = 1U;
                   if (VL_UNLIKELY((0x64U < vlSelf->__VactIterCount))) {
   #ifdef VL_DEBUG
                       Vt_gantt___024root___dump_triggers__act(vlSelf);
   #endif
                       VL_FATAL_MT("t/t_gen_alw.v", 6, "", "Active region did not converge.");
                   }
                   vlSelf->__VactIterCount = ((IData)(1U)`
                                              + vlSelf->__VactIterCount);
                   __VpreTriggered.andNot(vlSelf->__VactTriggered, vlSelf->__VnbaTriggered);
                   vlSelf->__VnbaTriggered.thisOr(vlSelf->__VactTriggered);
                   Vt_gantt___024root___eval_act(vlSelf);
               }
           }
           VL_EXEC_TRACE_ADD_RECORD(vlSymsp).sectionPop();
           if (vlSelf->__VnbaTriggered.any()) {
               __VnbaContinue = 1U;
               if (VL_UNLIKELY((0x64U < __VnbaIterCount))) {
   #ifdef VL_DEBUG
                   Vt_gantt___024root___dump_triggers__nba(vlSelf);
   #endif
                   VL_FATAL_MT("t/t_gen_alw.v", 6, "", "NBA region did not converge.");
               }
               __VnbaIterCount = ((IData)(1U) + __VnbaIterCount);
               Vt_gantt___024root___eval_nba(vlSelf);
           }
       }
       VL_EXEC_TRACE_ADD_RECORD(vlSymsp).sectionPop();
   }
```